### PR TITLE
Update pipelines

### LINKED ISSUE #64: update pypi publishing
https://blog.pypi.org/posts/2023-04-20-introducing-trusted-publishers/

### PR DESCRIPTION
**Pull request by AnthonyLim23**
This PR update the pipelines:

1. It allows us to test against Python 3.8, 3.9 and 3.10
2. It allows us to build against Python 3.8, 3.9 and 3.10
3. It now uses the PYPI trusted publisher

fixes #64 

To test:
1. check the automated tests are against all 3 expected versions
2. see that the repo has been updated https://pypi.org/project/quickBayes/#history

### CODE REVIEW

**Comment by AnthonyLim23** [comment]
https://docs.pypi.org/trusted-publishers/adding-a-publisher/

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -38,18 +38,19 @@ jobs:
...
-    # upload to PyPI on every tag starting with 'v'
-    # if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags/v')
-    # alternatively, to publish when a GitHub Release is created, use the following rule:
-    # if: github.event_name == 'release' && github.event.action == 'published'
+    permissions:
+      id-token: write
     steps:
+      - name: mint API token
+        id: mint
+        uses: tschm/token-mint-action@v1.0.2
```
Comment (RichardWaiteSTFC) [inline .github/workflows/release.yml:46]: This trusted publishing looks clever!

File: quickBayes-dev-win.yml [hunk 2]
```diff
@@ -4,8 +4,8 @@ channels:
   - conda-forge
 
 dependencies:
-  - python =3.8.*
-  - numpy =1.16.*
+  - python =3.10.*
```
Comment (RichardWaiteSTFC) [inline quickBayes-dev-win.yml:7]: So are these default values that get overwritten by the `matrix.python_version`?

File: .github/workflows/release.yml (hunk 1 above)
Comment (AnthonyLim23) [reply .github/workflows/release.yml:46]: It was a pain to get to work, the token mint thing was very helpful

File: quickBayes-dev-win.yml (hunk 2 above)
Comment (AnthonyLim23) [reply quickBayes-dev-win.yml:7]: This is generated by https://github.com/mantidproject/quickBayes/pull/67/files#diff-4ac0a1f77c600ccbb34e88eac122b3f4d9939799ea05ced9d552b80ee577c54cR17 which we run at https://github.com/mantidproject/quickBayes/pull/67/files#diff-c2e9f7c1bfaf65da79bbd29ed3ce389b2acc8b4099a257914a7292a66e6b9bc9R36 and the release pipeline does not use the conda recipe.

Comment (RichardWaiteSTFC) [reply quickBayes-dev-win.yml:7]: Ah OK I see, the `-dev-win.yml` etc. files are used only for testing - so does this mean tests are only ever run on python 3.10. Is it not a good idea to check for backwards compatibility of python versions, perhaps on the oldest python you support? I suppose there has to be a limit to how many tests you run though!

Comment (AnthonyLim23) [reply quickBayes-dev-win.yml:7]: we run all of the versions. The `.yml` files are generated by the python script ([link](https://github.com/mantidproject/quickBayes/pull/67/files#diff-4ac0a1f77c600ccbb34e88eac122b3f4d9939799ea05ced9d552b80ee577c54cR125)) and it takes the OS and python version as input to create the correct `yml` file. In https://github.com/mantidproject/quickBayes/pull/67/files#diff-c2e9f7c1bfaf65da79bbd29ed3ce389b2acc8b4099a257914a7292a66e6b9bc9R36 we loop over the python versions

Comment (RichardWaiteSTFC) [reply quickBayes-dev-win.yml:7]: OK thanks sorry I'm still fairly new to this github actions stuff! looks good!

Comment (AnthonyLim23) [reply quickBayes-dev-win.yml:7]: I did some black magic by creating scripts to generate conda recipes. It was the best way to make sure they are all in sync

**Pull request merged**